src/map.cpp

```diff
@@ -137,4 +138,93 @@ Map::~Map()
     close(mapfd_);
 }
 
+void MapManager::Add(std::unique_ptr<IMap> map)
+{
+  auto name = map->name_;
+  if (name.empty())
+    throw std::runtime_error("Map without name cannot be stored");
+
+  auto count = maps_by_name_.count(name);
+  if (count > 0)
+    throw std::runtime_error("Map with name: @" + name + " already exists");
+
+  std::shared_ptr<IMap> smap = std::move(map);
+  auto id = maps_by_id_.size();
+
+  maps_by_name_[name] = smap;
+  maps_by_id_.emplace_back(smap);
+
+  smap->id = id;
+}
+
+bool MapManager::Has(const std::string &name)
+{
+  return maps_by_name_.find(name) != maps_by_name_.end();
+}
+
+IMap &MapManager::Lookup(const std::string &name)
+{
+  auto search = maps_by_name_.find(name);
+  if (search == maps_by_name_.end())
+  {
+    throw std::runtime_error("No map with name @" + name + " defined");
+  }
+
+  return *search->second;
+}
+
+IMap &MapManager::Lookup(ssize_t id)
+{
+  return *maps_by_id_[id];
+}
+
+void MapManager::Set(MapManager::Type t, std::unique_ptr<IMap> map)
+{
+  maps_by_type_[t] = std::move(map);
+}
+
+IMap &MapManager::Lookup(Type t)
+{
+  auto search = maps_by_type_.find(t);
+  if (search == maps_by_type_.end())
+  {
+    throw std::runtime_error("No map of type " + to_string(t) + " defined");
+  }
+
+  return *search->second;
+}
+
+bool MapManager::Has(Type t)
+{
+  return maps_by_type_.find(t) != maps_by_type_.end();
+}
+
+void MapManager::Set(StackType t, std::unique_ptr<IMap> map)
+{
+  stackid_maps_[t] = std::move(map);
+}
+
+IMap &MapManager::Lookup(StackType t)
+{
+  return *stackid_maps_[t];
+}
+
+bool MapManager::Has(StackType t)
+{
+  return stackid_maps_.find(t) != stackid_maps_.end();
+}
+
+std::string to_string(MapManager::Type t)
+{
+  switch (t) {
+    case MapManager::Type::PerfEvent:
+      return "perf_event";
+    case MapManager::Type::Join:
+      return "join";
+    case MapManager::Type::Elapsed:
+      return "elapsed";
+    default:
+      throw std::runtime_error("BUG: Unknown MapManager::type");
```

Comment:
Since b20320b0 is in now
```suggestion
    
    return {}; // unreached
```